### PR TITLE
TagBox should use Chrome feature detection to specify RTL scroll behavior

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/core/utils/position.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import domAdapter from '../dom_adapter';
 import config from '../config';
 import typeUtils from '../utils/type';
 
+let cachedScrollBehavior;
```
Comment (San4es) [inline js/core/utils/position.js]: Use `callOnce` instead?

File: js/core/utils/position.js [hunk 2]
```diff
@@ -34,5 +37,34 @@ const getBoundingRect = (element) => {
...
+    const document = domAdapter.getDocument();
+
+    /* Append a RTL scrollable 1px square containing a 2px-wide child and check
+       the initial scrollLeft and whether it's possible to set a negative one.*/
+    document.body.insertAdjacentHTML('beforeend', `<div style='direction: rtl;
+       position: absolute; left: 0; top: 0; overflow: hidden; width: 1px;
+       height: 1px;'><div style='width: 2px; height: 1px;'></div></div>`);
+
+    const scroller = document.body.lastElementChild;
+    const initially_positive = scroller.scrollLeft > 0;
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
    const initiallyPositive = scroller.scrollLeft > 0;
```

File: js/core/utils/position.js [hunk 3]
```diff
@@ -34,5 +37,34 @@ const getBoundingRect = (element) => {
...
+    scroller.scrollLeft = -1;
+    const has_negative = scroller.scrollLeft < 0;
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
    const hasNegative = scroller.scrollLeft < 0;
```

File: js/core/utils/position.js [hunk 4]
```diff
@@ -34,5 +37,34 @@ const getBoundingRect = (element) => {
...
+
+    const result = {
+        'decreasing': has_negative || initially_positive,
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
        'decreasing': hasNegative || initiallyPositive,
```

File: js/core/utils/position.js [hunk 5]
```diff
@@ -34,5 +37,34 @@ const getBoundingRect = (element) => {
...
+        'positive': !has_negative
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
        'positive': !hasNegative
```

File: js/core/utils/position.js (hunk 2 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js (hunk 3 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js (hunk 4 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js (hunk 5 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js (hunk 1 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js [hunk 6]
```diff
@@ -34,5 +36,30 @@ const getBoundingRect = (element) => {
     return rect;
 };
 
+const getScrollBehavior = callOnce(function() {
```
Comment (San4es) [inline js/core/utils/position.js]:
```suggestion
const getScrollRtlBehavior = callOnce(function() {
```

File: js/core/utils/position.js [hunk 7]
```diff
@@ -34,5 +36,30 @@ const getBoundingRect = (element) => {
...
+    };
+
+    document.body.removeChild(scroller);
+
+    return result;
+});
+
 exports.getDefaultAlignment = getDefaultAlignment;
 exports.getBoundingRect = getBoundingRect;
+exports.getScrollBehavior = getScrollBehavior;
```
Comment (San4es) [inline js/core/utils/position.js]: Move to support.js? /cc @nightskylark

File: js/core/utils/position.js (hunk 6 above)
Comment (ksercs) [reply js/core/utils/position.js]: fixed

File: js/core/utils/position.js (hunk 7 above)
Comment (ksercs) [reply js/core/utils/position.js]: moved to `scrolRtlBehavior.js`

**Review by San4es — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**